contracts/reverseRegistrar/profiles/L2TextResolver.sol

```diff
@@ -0,0 +1,39 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.4;
+
+import "./L2ReverseResolverBase.sol";
+import "../../resolvers/profiles/ITextResolver.sol";
+
+abstract contract L2TextResolver is ITextResolver, L2ReverseResolverBase {
+    mapping(uint64 => mapping(bytes32 => mapping(string => string))) versionable_texts;
+
+    function _setText(
+        bytes32 node,
+        string calldata key,
+        string calldata value
+    ) internal {
+        versionable_texts[recordVersions[node]][node][key] = value;
+        emit TextChanged(node, key, key, value);
```

Comment:
Because L2 doesn't have a registry, registrar nor namewrapper, we don't have a way to decode the node when indexing. dm3's resolver changed [the behavior to pass DNS encoded name] (https://github.com/corpus-io/ENS-Bedrock-Resolver/blob/main/contracts/l2/profiles/text/TextResolver.sol#L19). Is it worth changing the interfaces for all `set*` functions to take `bytes calldata name`, instead of `bytes32 node` ? cc @Arachnid